base/regex.jl

```diff
@@ -120,7 +120,7 @@ function show(io::IO, m::RegexMatch)
     idx_to_capture_name = PCRE.capture_names(m.regex.regex)
     if !isempty(m.captures)
         print(io, ", ")
-        for i = 1:length(m.captures)
```

Comment:
Here, `idx_to_capture_name` is a `Dict` with `Int` keys, so we really need `i` to be an integer. Better keep the original code.